### PR TITLE
Frenet

### PR DESCRIPTION
**Pull request by eggrobin**

### CODE REVIEW

File: physics/dynamic_frame.hpp [hunk 1]
```diff
@@ -24,7 +37,51 @@ class DynamicFrame {
   virtual Vector<Acceleration, ThisFrame> GeometricAcceleration(
       Instant const& t,
       DegreesOfFreedom<ThisFrame> const& degrees_of_freedom) const = 0;
+
+  // The definition of the Frenet frame of a free fall trajectory in |ThisFrame|
+  // with the given |DegreesOfFreedom| at the given |Instant|.  Unless
```
Comment (pleroy) [inline physics/dynamic_frame.hpp]: |degrees_of_freedom|, |t|.  Better use the parameter names than the types.

File: physics/dynamic_frame.hpp [hunk 2]
```diff
@@ -24,7 +37,51 @@ class DynamicFrame {
...
+  // specialized otherwise by child classes, the member functions of the result
+  // fail when called at instants other than |t|.
+  virtual Rotation<Frenet<ThisFrame>, ThisFrame> FrenetFrame(
+      Instant const& t,
+      DegreesOfFreedom<ThisFrame> const& degrees_of_freedom) const;
+};
+
+// An inertial frame.
+template<typename OtherFrame, typename ThisFrame>
+class InertialFrame : public DynamicFrame<OtherFrame, ThisFrame> {
```
Comment (pleroy) [inline physics/dynamic_frame.hpp]: Assert that both frames are inertial?

File: physics/dynamic_frame.hpp [hunk 3]
```diff
@@ -24,7 +37,51 @@ class DynamicFrame {
...
+ public:
+  InertialFrame(Velocity<OtherFrame> const& velocity,
+                Position<OtherFrame> const& origin_at_epoch,
```
Comment (pleroy) [inline physics/dynamic_frame.hpp]: Why two parameters and not a DegreesOfFreedom?

File: physics/dynamic_frame.hpp [hunk 4]
```diff
@@ -24,7 +37,51 @@ class DynamicFrame {
   virtual Vector<Acceleration, ThisFrame> GeometricAcceleration(
```
Comment (pleroy) [inline physics/dynamic_frame.hpp]: At line 34, Frame->ThisFrame.

File: physics/dynamic_frame.hpp [hunk 5]
```diff
@@ -24,7 +37,51 @@ class DynamicFrame {
...
+                std::function<Vector<Acceleration, OtherFrame>(
+                    Instant const& t,
+                    Position<OtherFrame> const& q)> gravity);
+
+  RigidMotion<OtherFrame, ThisFrame> ToThisFrameAtTime(
+      Instant const& t) const override;
+  RigidMotion<ThisFrame, OtherFrame> FromThisFrameAtTime(
+      Instant const& t) const override;
+
+  // The acceleration due to the non-inertial motion of |Frame| and gravity.
```
Comment (pleroy) [inline physics/dynamic_frame.hpp]: What non-inertial motion?  Isn't this an inertial frame?  Also, there is no |Frame|.

File: physics/dynamic_frame_body.hpp [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+using geometry::Normalize;
+using geometry::R3x3Matrix;
+using geometry::Wedge;
+using quantities::Sqrt;
+using quantities::si::Metre;
+using quantities::si::Second;
+
+namespace physics {
+
+template <typename InertialFrame, typename ThisFrame>
```
Comment (pleroy) [inline physics/dynamic_frame_body.hpp]: No space after template.

File: physics/dynamic_frame_body.hpp [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+    Instant const& epoch,
+    OrthogonalMap<OtherFrame, ThisFrame> const& orthogonal_map,
+    std::function<Vector<Acceleration, OtherFrame>(
+        Instant const& t,
+        Position<OtherFrame> const& q)> gravity)
+    : velocity_(velocity),
+      origin_at_epoch_(origin_at_epoch),
+      epoch_(epoch),
+      orthogonal_map_(orthogonal_map),
+      gravity_(gravity) {}
```
Comment (pleroy) [inline physics/dynamic_frame_body.hpp]: std::move

File: physics/dynamic_frame.hpp [hunk 8]
```diff
@@ -24,7 +37,51 @@ class DynamicFrame {
...
+template<typename OtherFrame, typename ThisFrame>
+class InertialFrame : public DynamicFrame<OtherFrame, ThisFrame> {
+ public:
+  InertialFrame(Velocity<OtherFrame> const& velocity,
+                Position<OtherFrame> const& origin_at_epoch,
+                Instant const& epoch,
+                OrthogonalMap<OtherFrame, ThisFrame> const& orthogonal_map,
+                std::function<Vector<Acceleration, OtherFrame>(
+                    Instant const& t,
+                    Position<OtherFrame> const& q)> gravity);
```
Comment (pleroy) [inline physics/dynamic_frame.hpp]: Passing gravity here is very weird.  Could we pass an ephemeris instead?

File: physics/dynamic_frame_test.cpp [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+using Circular =
+    Frame<serialization::Frame::TestTag, serialization::Frame::TEST, true>;
+using Helical =
+    Frame<serialization::Frame::TestTag, serialization::Frame::TEST1, true>;
+
+Vector<Acceleration, Circular> Gravity(Instant const& t,
+                                       Position<Circular> const& q) {
+  Displacement<Circular> const r = q - Circular::origin;
+  auto const r2 = InnerProduct(r, r);
```
Comment (pleroy) [inline physics/dynamic_frame_test.cpp]: Use the Norm?

File: physics/dynamic_frame.hpp (hunk 1 above)
Comment (eggrobin) [inline physics/dynamic_frame.hpp]: Done.

File: physics/dynamic_frame.hpp (hunk 2 above)
Comment (eggrobin) [inline physics/dynamic_frame.hpp]: No, we don't want to require compile-time knowledge of inertialness.

File: physics/dynamic_frame.hpp (hunk 3 above)
Comment (eggrobin) [inline physics/dynamic_frame.hpp]: Good point.

File: physics/dynamic_frame.hpp (hunk 4 above)
Comment (eggrobin) [inline physics/dynamic_frame.hpp]: Done.

File: physics/dynamic_frame.hpp (hunk 5 above)
Comment (eggrobin) [inline physics/dynamic_frame.hpp]: Done.

File: physics/dynamic_frame_body.hpp (hunk 6 above)
Comment (eggrobin) [inline physics/dynamic_frame_body.hpp]: Doen.

File: physics/dynamic_frame_body.hpp (hunk 7 above)
Comment (eggrobin) [inline physics/dynamic_frame_body.hpp]: Done.

File: physics/dynamic_frame.hpp (hunk 8 above)
Comment (eggrobin) [inline physics/dynamic_frame.hpp]: Maybe, but then we'd really need dummy ephemerides so the tests remain palatable, and that's a job for another CL. `InertialFrame` exists for testing purposes only anyway.

File: physics/dynamic_frame_test.cpp (hunk 9 above)
Comment (eggrobin) [inline physics/dynamic_frame_test.cpp]: And then square it? That seems odd.

Comment (eggrobin) [inline physics/dynamic_frame_test.cpp]: On the other hand I could get rid of the explicit Sqrt(r2) using Normalize(r).

Comment (eggrobin) [inline physics/dynamic_frame_test.cpp]: That would just hide an additional division though, and I'm not sure it improves clarity.

File: physics/dynamic_frame_body.hpp [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+
+template<typename OtherFrame, typename ThisFrame>
+Vector<Acceleration, ThisFrame>
+InertialFrame<OtherFrame, ThisFrame>::GeometricAcceleration(
+    Instant const& t,
+    DegreesOfFreedom<ThisFrame> const& degrees_of_freedom) const {
+  return orthogonal_map_(
+      gravity_(t, FromThisFrameAtTime(t)(degrees_of_freedom).position()));
+}
+
```
Comment (pleroy) [inline physics/dynamic_frame_body.hpp]: Remove.

**Pull request merged**